openvdb/openvdb/version.h.in

```diff
@@ -102,45 +104,52 @@
     /// - @b openvdb::vX_YabiN::tree::Tree
     ///
     /// where X, Y and N are the major, minor and ABI version numbers, respectively.
-    #define OPENVDB_VERSION_NAME                                            \
-        OPENVDB_PREPROC_CONCAT(v,                                           \
-        OPENVDB_PREPROC_CONCAT(OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER,        \
-        OPENVDB_PREPROC_CONCAT(_, OPENVDB_LIBRARY_MINOR_VERSION_NUMBER)))
+    #define OPENVDB_VERSION_NAME v${OpenVDB_MAJOR_VERSION}_${OpenVDB_MINOR_VERSION}
 #else
     // This duplication of code is necessary to avoid issues with recursive macro expansion.
-    #define OPENVDB_VERSION_NAME                                            \
-        OPENVDB_PREPROC_CONCAT(v,                                           \
-        OPENVDB_PREPROC_CONCAT(OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER,        \
-        OPENVDB_PREPROC_CONCAT(_,                                           \
-        OPENVDB_PREPROC_CONCAT(OPENVDB_LIBRARY_MINOR_VERSION_NUMBER,        \
-        OPENVDB_PREPROC_CONCAT(abi, OPENVDB_ABI_VERSION_NUMBER)))))
+    #define OPENVDB_VERSION_NAME v${OpenVDB_MAJOR_VERSION}_${OpenVDB_MINOR_VERSION}abi${OPENVDB_ABI_VERSION_NUMBER}
 #endif
 
-/// @brief Library version number string of the form "<major>.<minor>.<patch>"
-/// @details This is a macro rather than a static constant because we typically
-/// want the compile-time version number, not the runtime version number
-/// (although the two are usually the same).
-/// @hideinitializer
-#define OPENVDB_LIBRARY_VERSION_STRING \
-    OPENVDB_PREPROC_STRINGIFY(OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER) "." \
-    OPENVDB_PREPROC_STRINGIFY(OPENVDB_LIBRARY_MINOR_VERSION_NUMBER) "." \
-    OPENVDB_PREPROC_STRINGIFY(OPENVDB_LIBRARY_PATCH_VERSION_NUMBER)
+/* Denotes whether VDB was built with IMath Half support */
+#cmakedefine OPENVDB_USE_IMATH_HALF
 
-/// @brief Library version number string of the form "<major>.<minor>.<patch>abi<abi>"
-/// @details This is a macro rather than a static constant because we typically
-/// want the compile-time version number, not the runtime version number
-/// (although the two are usually the same).
-/// @hideinitializer
-#define OPENVDB_LIBRARY_ABI_VERSION_STRING \
-    OPENVDB_LIBRARY_VERSION_STRING "abi" OPENVDB_PREPROC_STRINGIFY(OPENVDB_ABI_VERSION_NUMBER)
+/* Denotes whether VDB was built with Blosc support */
+#cmakedefine OPENVDB_USE_BLOSC
 
-/// Library version number as a packed integer ("%02x%02x%04x", major, minor, patch)
-#define OPENVDB_LIBRARY_VERSION_NUMBER \
-    ((OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER << 24) | \
-    ((OPENVDB_LIBRARY_MINOR_VERSION_NUMBER & 0xFF) << 16) | \
-    (OPENVDB_LIBRARY_PATCH_VERSION_NUMBER & 0xFFFF))
+/* Denotes whether VDB was built with ZLIB support */
+#cmakedefine OPENVDB_USE_ZLIB
```

Comment:
Added `OPENVDB_USE_BLOSC` and `OPENVDB_USE_ZLIB`.

Note that these defines do not technically need to be public defines, but I don't see any harm in adding them here. A lot of implementations I've see configure 2-3 files with defines like this and categorise them such:
 - 1) Config.h - Public configuration header intended to be queried by clients
 - 2) InternalConfig.h/InternalConfig.cmake - Internal configuration header/cmake used by the build system of the project but not installed.

These two variables falls into (2) because they are never used in any public header file. For example:
```
bool hasZlib(); // in header

// in .cc
bool hasZlib() { 
#ifdef OPENVDB_USE_ZLIB
return true;
#else
return false;
#endif
}
```

I can remove them and revert back to the `PRIVATE` define system if we think they shouldn't be exposed as such.

Reply:
Agreed, I was just playing devils advocate in that this information could also be provided outside the scope of the C++ headers (i.e. in via pkgconfig/cmake etc). Doing it this way would mean less "API" reliance on `#defines` which are hard to deprecate/remove in the future. Wanted to make sure this was clear in case there were any concerns.